tools/biolatency.py

```diff
@@ -193,6 +200,68 @@ def flags_print(flags):
         desc = "NoWait-" + desc
     return desc
 
+
+def _print_json_hist(vals, val_type, section_bucket=None):
+    hist_list = []
+    max_nonzero_idx = 0
+    for i in range(len(vals)):
+        if vals[i] != 0:
+            max_nonzero_idx = i
+    index = 1
+    prev = 0
+    for i in range(len(vals)):
+        if i != 0 and i <= max_nonzero_idx:
+            index = index * 2
+
+            list_obj = {}
+            list_obj['interval-start'] = prev
+            list_obj['interval-end'] = int(index) - 1
+            list_obj['count'] = int(vals[i])
+
+            hist_list.append(list_obj)
+
+            prev = index
+    histogram = {"ts": strftime("%Y-%m-%d %H:%M:%S"), "val_type": val_type, "data": hist_list}
+    if section_bucket:
+        histogram[section_bucket[0]] = section_bucket[1]
+    print(histogram)
+
+
+def print_json_hist(self, val_type="value", section_header="Bucket ptr",
+                       section_print_fn=None, bucket_fn=None, bucket_sort_fn=None):
```

Comment:
Looking at this function again. It has a lot of duplicates with print_log2_dist. It looks like what you suggested earlier make more sense. Let me implement print_json_hist as a library function. In the bcc python library (table.py), we can factor out common part of print_json_hist/print_log2_dist to avoid code duplication. WDYT?

Reply:
@yonghong-song Yes, that is why I was suggesting implementing the function in table.py. I had already implemented the function in table.py and can easily push it upstream so you won't have to write it again. WDYT?

Reply:
sounds good. thanks!

Reply:
@yonghong-song Looking carefully at `table.py`, I see that `print_linear_hist` also has a lot of overlap with `print_json_hist/print_log2_dist`. 
How about we factor out all of that in a separate PR? I can take it up.

Reply:
Sounds good. I will merge the current PR as you can have a followup to do proper refactoring. Thanks!